tools/parse_baselines/main.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Script used to calculate baselines from raw performance test output."""
+
+import argparse
+import os
+import tempfile
+import json
+from providers.types import FileDataProvider
+from providers.vsock import VsockDataParser
+from providers.network_tcp import NetworkTcpDataParser
+
+
+OUTPUT_FILENAMES = {
+    'vsock': 'test_vsock_throughput.out',
+    'network_tcp': 'test_network_tcp_throughput.out'
+}
+
+DATA_PARSERS = {
+    'vsock': VsockDataParser,
+    'network_tcp': NetworkTcpDataParser
+}
+
+
+def get_data_files(args):
+    """Return a list of files that contain results for this test."""
+    assert os.path.isdir(args.data_folder)
+
+    file_list = []
+
+    # Get all files in the dir tree that have the right name.
+    for root, _, files in os.walk(args.data_folder):
+        for file in files:
+            if file == OUTPUT_FILENAMES[args.test]:
+                file_list.append(os.path.join(root, file))
+
+    # We need at least one file.
+    assert len(file_list) > 0
+
+    return file_list
+
+
+def concatenate_data_files(data_files):
+    """Create temp file to hold all concatenated results for this test."""
+    outfile = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
My thought was to try something like the following:
```
>>> def boo():
...     x = tempfile.NamedTemporaryFile()
...     return x
... 
>>> def foo():
...     f = boo()
...     print(os.path.exists(f.name))
...     return f.name
... 
>>> os.path.exists(foo())
True
False
```

In our case, at the end of the `main()` function, the file will be effectively closed (part because the program finishes executing and part due to becoming out of scope and hence garbage collected, but I am not sure which one is the real motif), being deleted from the filesystem as well. It will prove the usage of the `unlink` unnecessary. Feel free to integrate it if you find it useful.

Reply:
great idea, integrated this. I was expecting that the interpreter would free up the memory and therefore close the file, if it sees that we only get the name prop from it.
However, it works if we save the outfile in a local variable and use that, as in your example